src/Monitor/Monitor/ScheduledQueryRules/UpdateScheduledQueryRuleCommand.cs

```diff
@@ -0,0 +1,138 @@
+{
+    /// <summary>
+    /// Updates a ScheduledQueryRule object
+    /// </summary>
+    [Cmdlet(VerbsData.Update, ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "ScheduledQueryRule",
+         SupportsShouldProcess = true, DefaultParameterSetName = ByRuleName), OutputType(typeof(PSScheduledQueryRuleResource))]
+    public class UpdateScheduledQueryRuleCommand : ManagementCmdletBase
+    {
+
+        private const string ByInputObject = "ByInputObject";
+        private const string ByRuleName = "ByRuleName";
+        private const string ByResourceId = "ByResourceId";
+
+        #region Cmdlet parameters
+
+        [Parameter(ParameterSetName = ByInputObject, Mandatory = true, ValueFromPipeline = true,
+            HelpMessage = "The Scheduled Query Rule resource")]
+        [ValidateNotNull]
+        public PSScheduledQueryRuleResource InputObject { get; set; }
+
+        [Parameter(ParameterSetName = ByResourceId, Mandatory = true, ValueFromPipelineByPropertyName = true,
+            HelpMessage = "The resource Id")]
+        [ValidateNotNullOrEmpty]
+        public string ResourceId { get; set; }
+
+        /// <summary>
+        /// Alert name
+        /// </summary>
+        [Parameter(ParameterSetName = ByRuleName, Mandatory = true, ValueFromPipelineByPropertyName = true,
+            HelpMessage = "The alert name")]
+        [ResourceNameCompleter("Microsoft.insights/scheduledqueryrules", nameof(ResourceGroupName))]
+        [ValidateNotNullOrEmpty]
+        public string RuleName { get; set; }
+
+        /// <summary>
+        /// The resource group name
+        /// </summary>
+        [Parameter(ParameterSetName = ByRuleName, Mandatory = true, ValueFromPipelineByPropertyName = true,
+            HelpMessage = "The resource group name")]
+        [ResourceGroupCompleter]
+        [ValidateNotNullOrEmpty]
+        public string ResourceGroupName { get; set; }
+
+        /// <summary>
+        /// Alert status - enabled or not, supported values - "true", "false"
+        /// </summary>
+        [Parameter(ParameterSetName = ByRuleName, Mandatory = true,
+            HelpMessage = "The azure alert state - valid values - true, false")]
+        [Parameter(ParameterSetName = ByInputObject, Mandatory = true,
+            HelpMessage = "The azure alert state - valid values - true, false")]
+        [Parameter(ParameterSetName = ByResourceId, Mandatory = true,
+            HelpMessage = "The azure alert state - valid values - true, false")]
+        [ValidateSet("true", "false")]
+        [PSArgumentCompleter("true", "false")]
+        public string Enabled { get; set; }
+
+        #endregion
+
+        protected override void ProcessRecordInternal()
+        {
+            ResourceIdentifier resourceIdentifier = null;
+            ScheduledQueryRuleResource resource = null;
+
+            // ByInputObject parameter set
+            if (this.IsParameterBound(c => c.InputObject) || this.InputObject != null)
```

Comment:
@shpimpal same comment in your cmdlets about removing additional null checks for parameters that have a validation attribute

Reply:
Had to add the check as the parameter was not being binded when invoked from mock tests.